src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs

```diff
@@ -0,0 +1,619 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Generic;
+using System.Linq;
+using WinRT.SourceGenerator;
+
+namespace Generator 
+{
+    public class WinRTRules
+    {
+
+        #region ModifierHelpers
+        private bool SyntaxTokenIs(SyntaxToken stx, string str) { return stx.Value.Equals(str); }
+        
+        private bool ModifiersContains(SyntaxTokenList modifiers, string str)
+        {
+            foreach (var modifier in modifiers)
+            {
+                if (SyntaxTokenIs(modifier, str)) { return true; }
+            }
+            return false;
+        }
+
+        private bool PropertyIsPublic(PropertyDeclarationSyntax p) { return ModifiersContains(p.Modifiers, "public"); }

```

Comment:
I think all these have the same base class of MemberDeclarationSyntax which you might be able to use to simplify this to 1 function.

Reply:
Good idea!